### PR TITLE
added a solution for palindrome homework

### PR DESCRIPTION
**Pull request by Prox1k**

### CODE REVIEW

File: src/main/java/core/basesyntax/Palindrome.java [hunk 1]
```diff
@@ -21,6 +21,10 @@
      * <p>Результат: true</p>
      */
     public boolean isPalindrome(String text) {
-        return false;
+        StringBuilder clean = new StringBuilder(text.replaceAll("[^a-zA-Z0-9]", "").toLowerCase());
+        String cleanStr = clean.toString();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Palindrome.java]:
```suggestion
        String cleanText = new StringBuilder(text.replaceAll("[^a-zA-Z0-9]", "").toLowerCase()).toString();
```

File: src/main/java/core/basesyntax/Palindrome.java [hunk 2]
```diff
@@ -21,6 +21,10 @@
...
+        String cleanReversed = clean.reverse().toString();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Palindrome.java:26]:
```suggestion
        String textReversed = clean.reverse().toString();
```

File: src/main/java/core/basesyntax/Palindrome.java [hunk 3]
```diff
@@ -21,6 +21,10 @@
...
+        boolean res = cleanStr.equals(cleanReversed);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Palindrome.java]: do we need boolean variable here?
we can write return ...

File: src/main/java/core/basesyntax/Palindrome.java (hunk 1 above)
Comment (Prox1k) [reply src/main/java/core/basesyntax/Palindrome.java]: You suggest to make cleanText as String, but we will need StringBuilder to use reverse() method in your next suggestion. Am I right?

File: src/main/java/core/basesyntax/Palindrome.java (hunk 2 above)
Comment (Prox1k) [reply src/main/java/core/basesyntax/Palindrome.java:26]: Here if I use String cleanText = new StringBuilder(...) ; then I might not have a possibility to use cleanText.reverse(); , I would need to create StringBuilder, what I did in the beggining of my code. Right?

File: src/main/java/core/basesyntax/Palindrome.java (hunk 1 above)
Comment (Sofasmile) [reply src/main/java/core/basesyntax/Palindrome.java]: but rename the variable

File: src/main/java/core/basesyntax/Palindrome.java (hunk 2 above)
Comment (Sofasmile) [reply src/main/java/core/basesyntax/Palindrome.java:26]: ok, but rename the variable

File: src/main/java/core/basesyntax/Palindrome.java (hunk 3 above)
Comment (Sofasmile) [reply src/main/java/core/basesyntax/Palindrome.java]: and what about this comment?

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**